src/durableorchestrationcontext.ts

```diff
@@ -65,7 +65,7 @@ export class DurableOrchestrationContext {
      * @returns A Durable Task that completes when the called activity
      * function completes or fails.
      */
-    public callActivity(name: string, input?: unknown): Task {
+    public callActivity<T>(name: string, input?: T): Task {

```

Comment:
I'm a little concerned by the confusion this will cause in comparison with C#, where `CallActivityAsync<T>` has `T` refer to the `return` type, not the input type.

Unfortunately, with our use of generator functions, we don't have a way to cleanly specify a type for the return value.

Reply:
I'll admit I haven't looked at the C#/.NET API for a long time, and wasn't aware of how the type system there was represented (I didn't realise that `input` is `object` there, I assumed it would've been typed).

Admittedly, a lot of this PR was originally written around the Entities site of durable functions, not orchestrators, as I was trying to get them strongly typed so you can do stuff like I demo in https://github.com/Azure/azure-functions-durable-js/pull/195/files/8dc1bf93ebfda83db5ee8bda88764ba18e09e371#diff-efcc2398105c3d67a41a35488e1f984e83a7d96923c24c07bb9fa4e84a2c3fb6

Then I figured I'd apply the same pattern on orchestrators, came across the limitation with `yield` and went "eh, it's not breaking anything" so left it in.

Reply:
I think orchestrations are a bit of a mess right now typing wise, and until we find a way around our current `yield` typing, I think it may be best to leave them as is.

I think the entity stuff we should definitely move forward with.

I'm willing to budge on the orchestration stuff if it feels sufficiently "typescripty", but I personally feel the cons of the generic typing outway the pros. 

That being said, it still makes perfect sense for `GetInput` I would say!